lib/Doctrine/ORM/Cache/DefaultQueryCache.php

```diff
@@ -99,7 +99,7 @@ public function get(QueryCacheKey $key, ResultSetMapping $rsm, array $hints = []
         foreach ($cacheEntry->result as $index => $entry) {
             $entityEntry = is_array($entries) ? ($entries[$index] ?? null) : null;
 
-            if ($entityEntry === null) {
+            if (!$entityEntry) {
```

Comment:
This should rather be `! $entityEntry instanceof CacheEntry` (or something more specific to this usage)